### ISSUE TITLE
Ciw Doesn't Complain about missing Transition_matrices field

### ISSUE THREAD

**Issue by alcarney** [CiwPython/Ciw#79]
When giving a parameter dictionary to Ciw it doesn't explain that a Transition_matrix should be given

**Comment by geraintpalmer** [comment]
Rolling this issue into another enhancement feature: use Cerberus (https://github.com/nicolaiarocci/cerberus) for input validation, and properly tidy up the  validation side of Ciw.

**Comment by alcarney** [comment]
Probably something you will be able to do once you've hooked up cerberus but I thought it would be worth mentioning that error messages in ciw are quite cryptic at the moment. For example I'm using the following parameter dictionary

```
params = {
    'Arrival_distributions' : {'Class 0': [[ 'Uniform', 0.3, 0.6], ['NoArrivals']],
                               'Class 1': [[ 'Uniform', 0.4, 0.8], ['NoArrivals']],
                               'Class 2': [['NoArrivals'], ['Uniform', 0.1, 0.5 ]],
                               'Class 3': [['NoArrivals'], ['Uniform', 0.2, 0.6 ]]},
    'Number_of_nodes': 2,
    'north_south': [[25, 1], [50, 0]],
    'east_west': [[25, 0], [50, 1]],
    'Number_of_servers': ['north_south', 'east_west'],
    'Queue_capacities': ['Inf', 'Inf'],
    'Number_of_classes': 4,
    'Service_distributions': {'Class 0': [['Deterministic', 0.0], ['Deterministic', 0.0]],
                              'Class 1': [['Deterministic', 0.0], ['Deterministic', 0.0]],
                              'Class 2': [['Deterministic', 0.0], ['Deterministic', 0.0]],
                              'Class 3': [['Deterministic', 0.0], ['Deterministic', 0.0]]},
    'Transition_matrices': {'Class 0': [[0.0, 0.0], [0.0, 0.0]],
                            'Class 1': [[0.0, 0.0], [0.0, 0.0]],
                            'Class 2': [[0.0, 0.0], [0.0, 0.0]],
                            'Class 3': [[0.0, 0.0], [0.0, 0.0]]}
}
```

And the error I get from ciw is:

```
ValueError: Ensure that valid Arrival and Service Distributions are used.
```
Which leaves me to having to guess where my error is and what might be wrong. It would be useful if ciw said which distribution(s) it is not recognising

**Comment by geraintpalmer** [comment]
Hi @alcarney if you want no Arrivals, then put 'NoArrivals' (without the square brackets.

e.g. try:
```
     'Arrival_distributions' : {'Class 0': [[ 'Uniform', 0.3, 0.6], 'NoArrivals'],
```

This is something that will be sorted once I get around to tidying up the parameter validation stuff. Also noticed this isn't explained very well in the documentation. Thanks for pointing this out!

**Comment by alcarney** [comment]
Great thanks! :) Thought I was going mad :p

**Comment by geraintpalmer** [comment]
Addressed in #114

**Comment by drvinceknight** [comment]
Can we close this issue?

**Comment by geraintpalmer** [comment]
Addressed in #114
